src/components/views/LocalModList.vue

```diff
@@ -438,47 +443,43 @@ import SearchUtils from '../../utils/SearchUtils';
                 this.$emit('error', updatedList);
                 return updatedList;
             }
-            await this.$store.dispatch("updateModList", updatedList);
-            const err = await ConflictManagementProvider.instance.resolveConflicts(updatedList, this.contextProfile!);
-            if (err instanceof R2Error) {
-                this.$emit('error', err);
-            }
-            this.filterModList();
+            await this.updateModListAfterChange(updatedList);
         }
 
         async uninstallMod(vueMod: any) {
             let mod: ManifestV2 = new ManifestV2().fromReactive(vueMod);
             try {
                 for (const dependant of Dependants.getDependantList(mod, this.modifiableModList)) {
-                    const result = await this.performUninstallMod(dependant);
+                    this.modBeingUninstalled = dependant.getName();
+                    const result = await this.performUninstallMod(dependant, false);
                     if (result instanceof R2Error) {
                         this.$emit('error', result);
+                        this.modBeingUninstalled = null;
                         return;
                     }
                 }
-                const result = await this.performUninstallMod(mod);
+                this.modBeingUninstalled = mod.getName();
+                const result = await this.performUninstallMod(mod, false);
                 if (result instanceof R2Error) {
                     this.$emit('error', result);
+                    this.modBeingUninstalled = null;
                     return;
                 }
             } catch (e) {
                 // Failed to uninstall mod.
                 const err: Error = e as Error;
                 this.$emit('error', err);
+                this.modBeingUninstalled = null;
                 LoggerProvider.instance.Log(LogSeverity.ACTION_STOPPED, `${err.name}\n-> ${err.message}`);
             }
             this.selectedManifestMod = null;
+            this.modBeingUninstalled = null;
             const result: ManifestV2[] | R2Error = await ProfileModList.getModList(this.contextProfile!);
             if (result instanceof R2Error) {
                 this.$emit('error', result);
                 return;
             }
-            await this.$store.dispatch("updateModList", result);
-            const err = await ConflictManagementProvider.instance.resolveConflicts(result, this.contextProfile!);
-            if (err instanceof R2Error) {
-                this.$emit('error', err);
-            }
-            this.filterModList();
+            await this.updateModListAfterChange(result);
```

Comment:
Looks like there are a handful of code paths here which could in theory apply changes without refreshing the mod list, would it make sense to always refresh it before returning?

I'll push that change on top of this before merging since it seems a bit less prone to state management bugs that way.

Reply:
Done in b64c89cf306488d50b3e846aae32bbd9a5ae32a7